app/models/product.rb

```diff
@@ -2,20 +2,28 @@ class Product < ApplicationRecord
   has_many :line_items
   has_many :orders, through: :line_items
 
-  validates :title, :description, :image_url, presence: { message: "can't be blank"}
-  validates :title, uniqueness: true
-  validates :image_url, allow_blank: true, format: {with: %r{\.(gif|jpg|png)\z}i, message: 'must be a URL for GIF, JPG or PNG image.'}
-  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  with_options presence: true do
+    validates :url, url: true
+    validates :title, :description, :permalink
+    validates :title, uniqueness: true
```

Comment:
does this not add presence validation twice on title?
Please read about what with_options does